src/Microsoft.ML.OnnxTransform/OnnxCatalog.cs

```diff
@@ -11,23 +11,53 @@ public static class OnnxCatalog
     {
         /// <summary>
         /// Initializes a new instance of <see cref="OnnxScoringEstimator"/>.
+        /// All column names are provided, the input data column names/types must exactly match
+        /// all model input names. All possible output columns are then generated, with names/types
+        /// specified by the model.
         /// </summary>
         /// <param name="catalog">The transform's catalog.</param>
         /// <param name="modelFile">The path of the file containing the ONNX model.</param>
-        /// <param name="outputColumnNames">The input columns.</param>
-        /// <param name="inputColumnNames">The output columns resulting from the transformation.</param>
+        /// <param name="gpuDeviceId">Optional GPU device ID to run execution on. Null for CPU.</param>
+        /// <param name="fallbackToCpu">If GPU error, raise exception or fallback to CPU.</param>
         public static OnnxScoringEstimator ApplyOnnxModel(this TransformsCatalog catalog,
             string modelFile,
-            string[] outputColumnNames,
-            string[] inputColumnNames)
-        => new OnnxScoringEstimator(CatalogUtils.GetEnvironment(catalog), outputColumnNames, inputColumnNames, modelFile);
+            int? gpuDeviceId = null,
+            bool fallbackToCpu = false)
+        => new OnnxScoringEstimator(CatalogUtils.GetEnvironment(catalog), modelFile, gpuDeviceId, fallbackToCpu);
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="OnnxScoringEstimator"/>.
+        /// </summary>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="modelFile">The path of the file containing the ONNX model.</param>
+        /// <param name="outputColumnName">The input column.</param>
+        /// <param name="inputColumnName">The output column resulting from the transformation.</param>
+        /// <param name="gpuDeviceId">Optional GPU device ID to run execution on. Null for CPU.</param>
+        /// <param name="fallbackToCpu">If GPU error, raise exception or fallback to CPU.</param>
+        public static OnnxScoringEstimator ApplyOnnxModel(this TransformsCatalog catalog,
+            string modelFile,
+            string outputColumnName,
```

Comment:
shouldn't both of the ApplyOnnxModel APIs take a tuple of (OutputColumnName, InputColumnName) to be consistent with the other estimators?

Reply:
They cannot, the reason is that the OnnxTransform is not a `OneToOneTransformer`. So there could be a different number of inputs and outputs.

---
In reply to: [253715453](https://github.com/dotnet/machinelearning/pull/2367#discussion_r253715453) [](ancestors = 253715453)